### PR TITLE
[ZEPPELIN-551] Add paragraph scope for angular object

### PR DESCRIPTION
**Pull request by Leemoonsoo**
### What is this PR for?

Add paragraph scope for angular object. While it changes some internal api and ZeppelinServer - Interpreter process protocol (thrift), it's better be merged after creating 0.5.6 release branch

### What type of PR is it?
Improvement

### Is there a relevant Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-551

### How should this be tested?
Creating AngularObject now taking 'paragraphId' as a parameter in addition to 'noteId'.
When 'paragraphId' is null, the AngularObject becomes notebook scope, otherwise it becomes paragraph scope. 

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? Yes

Incompatible with interpreter binary that built with older version because of this PR updates thrift idl

* Does this needs documentation? internal api change

### CODE REVIEW

**Comment by Leemoonsoo** [comment]
ready to merge

**Comment by corneadoug** [comment]
I will take a look at it today then :)

**Comment by corneadoug** [comment]
Do you have some example or documentation on how you can bind at paragraph or notebook level from UI?

**Comment by Leemoonsoo** [comment]
Let me update javadoc

**Comment by Leemoonsoo** [comment]
@corneadoug updated javadoc

**Comment by corneadoug** [comment]
Thanks, I will look at it

File: testing/startSparkCluster.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ if [ ! -d "${SPARK_HOME}" ]; then
     echo "${SPARK_VERSION}" | grep "^1.[12].[0-9]" > /dev/null
     if [ $? -eq 0 ]; then
         # spark 1.1.x and spark 1.2.x can be downloaded from archive
-        wget http://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz
+        wget -q http://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz
```
Comment (felixcheung) [inline testing/startSparkCluster.sh]: this will help reduce travis log length quite a bit!

**Comment by felixcheung** [comment]
looks good

**Comment by Leemoonsoo** [comment]
Thanks @felixcheung taking a look. I'm merging it into master.

**Pull request closed without merging**